lib/matplotlib/figure.py

```diff
@@ -3714,3 +3720,45 @@ def figaspect(arg):
     # the min/max dimensions (we don't want figures 10 feet tall!)
     newsize = np.clip(newsize, figsize_min, figsize_max)
     return newsize
+
+
+def _parse_figsize(figsize, dpi):
+    """
+    Convert a figsize expression to (width, height) in inches.
+
+    Parameters
+    ----------
+    figsize : (float, float) or (float, float, str)
+        This can be
+
+        - a tuple ``(width, height, unit)``, where *unit* is one of "inch", "cm",
+          "px".
+        - a tuple ``(x, y)``, which is interpreted as ``(x, y, "inch")``.
+
+    dpi : float
+        The dots-per-inch; used for converting 'px' to 'inch'.
+    """
+    num_parts = len(figsize)
+    if num_parts == 2:
+        return figsize
+    elif num_parts == 3:
+        x, y, unit = figsize
+        if unit == 'inch':
```

Comment:
```suggestion
        if 'in' in unit:
```

Reply:
We have to be precise. We don't want to match e.g. "point" here
```suggestion
        if unit in ('inch', 'in'):
```